### PR TITLE
Update `Permute` operation tests

### PR DESCRIPTION
**Pull request by antalszava**
**Context**

There are test cases for the `Permute` operation that involves creating a QNode and checking that the underlying tape is expanded. Due to the tape expansion, the `Permute` operation is decomposed into `SWAP` operations.

These test cases call the `QuantumTape.expand` method explicitly. However, as the test cases use a QNode, and `default.qubit`, the device at hand doesn't support the `Permute` operation, tape expansion happens internal. Therefore, by the time the second call to the `QuantumTape.expand` method is placed, the tape has already been expanded.

This second call at the moment leaves the tape as is. However, defining a decomposition for `qml.SWAP` (as is done in #1317) would result in another depth of tape expansion and in [causing errors to the current test cases](https://github.com/PennyLaneAI/pennylane/pull/1317/checks?check_run_id=2587487045#step:10:279) which assume the tape contains `qml.SWAP` operations.

**Changes**

Updates the tests creating a QNode to not call `QuantumTape.expand`.

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -157,6 +157,10 @@ random_mat2 = rng.standard_normal(3, requires_grad=False)
 
 <h3>Improvements</h3>
 
+* Tests for the `Permute` operation are adjusted so that tape expansion happens
+  once.
```
Comment (josh146) [inline .github/CHANGELOG.md:161]: I'm not so sure this belongs in 'improvements', as it is a bugfix to the tests suite (and not an improvement in PL directly). Perhaps worth moving to Bug fix, and adding a bit more detail here?

**Review by josh146 — APPROVED** [review]
Thanks @antalszava for both finding this _and_ fixing this!

Does it make sense for `Permute` to be a native `default.qubit` operation in the future? Potentially this might speed up simulations which use it.

File: .github/CHANGELOG.md (hunk 1 above)
Comment (antalszava) [reply .github/CHANGELOG.md]: Oh, good point! Updated

**Comment by antalszava** [comment]
Thank you for the comments @josh146! Created an issue for native support of `Permute` in `default.qubit` and updated the changelog.

**Pull request merged**